src/Core/Containers/VectorArray.hpp

```diff
@@ -3,21 +3,34 @@
 #include <Core/Containers/AlignedStdVector.hpp>
 #include <Core/RaCore.hpp>
 
+#include <type_traits> // std::is_integral
+
 namespace Ra {
 namespace Core {
 
+/// Internal class used to compute types and values used by VectorArray, to handle both
+/// eigen matrices and integral types
+template <typename V, bool isIntegral>
+struct VectorArrayTypeHelper {};
+
 /// This class is a wrapper around a std::vector of Core::Vectors.
 /// which allow to use the stdlib's dynamic array implementation, yet pass it as
 /// a matrix when Eigen needs it with the getMap() method.
 template <typename V>
 class VectorArray : public AlignedStdVector<V>
 {
+  private:
+    using TypeHelper =
+        VectorArrayTypeHelper<V, std::is_integral<V>::value || std::is_floating_point<V>::value>;
```

Comment:
```suggestion
        VectorArrayTypeHelper<V, std:: is_scalar_v<V>>;
```
(see https://en.cppreference.com/w/cpp/types/is_scalar)
This will allow to have char* attributes (eg C strings), enum values and others as attribute values.